iam/src/procedures/index.ts

```diff
@@ -10,19 +11,40 @@ export type GenerateTwitterAuthUrlRequestBody = {
   callback: string;
 };
 
+export type GenerateGithubAuthUrlRequestBody = {
+  callback: string;
+};
+
 export type GenerateBrightidBody = {
   contextIdData: string;
 };
 
 router.post("/twitter/generateAuthUrl", (req: Request, res: Response): void => {
   const { callback } = req.body as GenerateTwitterAuthUrlRequestBody;
   if (callback) {
-    const state = getSessionKey();
-    const client = initClient(callback, state);
+    const state = twitterOAuth.getSessionKey();
+    const client = twitterOAuth.initClient(callback, state);
+
+    const data = {
+      state,
+      authUrl: twitterOAuth.generateAuthURL(client, state),
+    };
+
+    res.status(200).send(data);
+  } else {
+    res.status(400);
+  }
+});
+
+router.post("/github/generateAuthUrl", (req: Request, res: Response): void => {
+  const { callback } = req.body as GenerateGithubAuthUrlRequestBody;
+  if (callback) {
+    const state = githubOAuth.getSessionKey();
+    const clientId = process.env.GITHUB_CLIENT_ID;
 
     const data = {
       state,
-      authUrl: generateAuthURL(client, state),
+      authUrl: `https://github.com/login/oauth/authorize?client_id=${clientId}&redirect_uri=${callback}&state=${state}`,
```

Comment:
this URL could be generated entirely on the front-end; the query parameters are all public info since the `client_secret` isn't required here. 
maybe even the `state` could be generated & checked on the front-end since the oauth redirect opens a new window, and the current page should(?) still have the initial `state` from the auth url